src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/ZipStatusController.java

```diff
@@ -24,7 +27,15 @@ public ZipStatusController(ZipFileStatusService service) {
     // endregion
 
     @GetMapping
-    public ZipFileStatus findByFileName(@RequestParam("name") String fileName) {
-        return service.getStatusFor(fileName);
+    public ResponseEntity<ZipFileStatus> getStatus(
+        @RequestParam(value = "name", required = false) Optional<String> fileName,
+        @RequestParam(value = "ccd_id", required = false) Optional<String> ccdId) {
+
+        if (fileName.isPresent() && !ccdId.isPresent()) {
+            return ResponseEntity.ok(service.getStatusFor(fileName.get()));
```

Comment:
The service method should be renamed to `getStatusByName` (or `getStatusByFileName`) - similar to `getStatusByCcdId`).

Reply:
It has been done in another branch.